### ISSUE TITLE
info_dict and info_table on Dataset should handle not-yet-defined counts

### ISSUE THREAD

**Issue by tstolarczyk** [gammapy/gammapy#3109]
based on 0.17

I suggest that `info_dict `and `info_table` handle the fact that `counts `do not exist (e.g. in a simulation before `dataset.fake()`).
Possible implementation (to be done for all related values):
```
if (counts != None):
   info["n_on"] = self.counts.data[mask].sum()
else:
    info['n_on'] = float('NaN')
```
Thx

**Comment by adonath** [comment]
Thanks a lot @tstolarczyk , as far as I can tell the issue  has already been fixed and will be available in v0.18 https://github.com/gammapy/gammapy/blob/master/gammapy/datasets/map.py#L1127

Here is the relevant code in v0.17: 
https://github.com/gammapy/gammapy/blob/v0.17/gammapy/datasets/spectrum.py#L623

**Comment by adonath** [comment]
The issue is fixed in `v0.18.2`.
